### DISCUSSION TITLE
Fluentd interfaces status check

### DISCUSSION THREAD

**Discussion by mrudrego** [fluent/fluentd#3967] [General]
Fluentd can connect to multiple interfaces based on the plugins and their configuration. Connectivity issues to these interfaces can be found via logs.  Is there any other way like API or any other means to get information of broken connection so this can be used to decipher the status of connection ?

Thanks,

**Comment by fujimotos** [comment]
You can use `@FLUENT_LOG` to transmit Fluentd's log to arbitrary
server. For details, please see:

https://docs.fluentd.org/deployment/logging#capture-fluentd-logs

**Reply by mrudrego** [reply]
Thanks for the reply.
@FLUENTD_LOG would help in transmitting all the fluentd's log but is there any API or other means to know only about the interfaces which fluentd connects to. With this we can raise alarms to take necessary actions on the interfaces ?

Thanks,

**Reply by fujimotos** [reply]
`in_monitor_agent`  exposes a basic statistics over HTTP. See the
official doc for the exposed stats:

https://github.com/fluent/fluentd/blob/master/lib/fluent/plugin/in_monitor_agent.rb

`in_prometheus` is an another option. You can retrieve Prometheus-
compatible outputs via HTT.

https://docs.fluentd.org/monitoring-fluentd/monitoring-prometheus 

Neither exposes raw td-agent logs, though (you need to use `@FLUENT_LOG`
for that purpose). So choose the suitable one according to your needs.
